### PR TITLE
mesa-dri: enable vc4 driver for armv6l

### PR DESCRIPTION
**Pull request by ericonr**
Related to #20861

### CODE REVIEW

**Comment by xtraeme** [comment]
missing revbump

**Comment by ericonr** [comment]
@xtraeme fixed! Given that xwayland wasn't built for armv6l because of the lack of VC4 driver (at least that's what 85e2e69d439 seems to indicate), can I add it to the xorg-server template and include the commit in this PR?

**Comment by xtraeme** [comment]
Sure

**Comment by ericonr** [comment]
Given that I'd be enabling a new package to be built, it shouldn't require a revbump, right?

**Comment by xtraeme** [comment]
you are wrong, it's the opposite! you need to revbump when the binpkgs change somehow.

**Comment by ericonr** [comment]
@xtraeme updated all the necessary packages and added the arch checks. I don't know if there's anywhere I could fit this snippet to avoid duplication, but if there is, it would probably be a good idea. Also added revbumps for everybody, I think.

File: srcpkgs/xorg-server/template [hunk 1]
```diff
@@ -34,7 +34,7 @@ build_options="elogind"
 desc_option_elogind="Rootless Xorg support with elogind"
 
 case "$XBPS_TARGET_MACHINE" in
```
Comment (st3r4g) [inline srcpkgs/xorg-server/template:36]: I wonder if this case statement could be removed altogether. All officially supported arches are included, and ppc too. Is this important for some obscure arch?

Comment (ericonr) [reply srcpkgs/xorg-server/template:36]: It excludes MIPS, right?

Comment (st3r4g) [reply srcpkgs/xorg-server/template:36]: yeah, mips and other arm* (`./xbps-src -h` gives a list)

Comment (ericonr) [reply srcpkgs/xorg-server/template:36]: What I meant is that MIPS is also officially supported (at least that's what it says at voidlinux.org), so we have to consider that when building these packages.

Comment (travankor) [reply srcpkgs/xorg-server/template:36]: I don't think void builds packages for MIPS... also MIPS as an architecture is going the way of the dodo thanks to RISCV

Comment (xtraeme) [reply srcpkgs/xorg-server/template:36]: See http://repo.voidlinux.de by @pullmoll

File: srcpkgs/gdm/template [hunk 2]
```diff
@@ -43,6 +42,11 @@ if [ "$CROSS_BUILD" ]; then
 	hostmakedepends+=" glib-devel pam-devel"
 fi
 
+case "$XBPS_TARGET_MACHINE" in
```
Comment (travankor) [inline srcpkgs/gdm/template:45]: missing comment above

File: srcpkgs/xorg-server/template (hunk 1 above)
Comment (travankor) [reply srcpkgs/xorg-server/template:36]: Technically, gallium-pipe is hardware agnostic and will support the obscure arches. It's not very efficient, though, but maybe there will be a better software renderer?

Comment (ericonr) [reply srcpkgs/xorg-server/template:36]: @xtraeme do you think keeping this check for MIPS is the best way of doing it?

File: srcpkgs/gdm/template (hunk 2 above)
Comment (ericonr) [reply srcpkgs/gdm/template:45]: thanks :)

**Comment by ericonr** [comment]
@q66 could you take a look at this PR, please? I rebased it and I think it's as close to ready as it can be, unless we can determine for sure that we can remove the `xorg-server-xwayland` check.

**Comment by q66** [comment]
let's drop the xwayland check, and enable xwayland in xorg everywhere, how about that?

**Pull request merged**